Src/PerformanceCollector/Perf.Shared/Implementation/QuickPulse/Helpers/QuickPulseTaskModuleScheduler.cs

```diff
@@ -1,54 +0,0 @@
-﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.Helpers
```

Comment:
no issue happened on functions v2 because we used tasks. Now we are switching to threads ad it may start happening.

The reason for threads is we pretend it does guarantee QP to run during thread pool starvation. But you knbow what - QP runs on top of HttpClient and even tough it's a spearate thread, HttpClient  methods are async and run in thread pool threads.

So I guess the whole intent to run QP ina separate thread does not make sense and I'd rather prefer for everything to run on top of tasks.

Reply:
I believe the original reason why Threads were used was to ensure QP was able to accumulate data and aggregate in fixed intervals. Sending data to backend using httpclient maybe delayed, but the data would be aggregated in the way expected by QP backend. Similar reasoning was used in MetricAggregator as well to get dedicated thread, to ensure aggregation occur in fixed interval.

@tokaplan Do you have any additional thoughts here? Originaly QP for .net core didn't use Threads because threads nuget was not referenced in the project.

Reply:
@netgh0st as well to share any additional comments. I think @netgh0st did the .net core support.

Reply:
Story time.

Once upon a time, around 4am, I was leaning forward in my chair, staring at a glaring screen. After quite a few weeks of designing, prototyping, and redesigning, I had finally finished the very first version of QuickPulse! All telemetry items passing through the SDK pipeline were being registered, accumulated, and all metrics were being aggregated correctly. The only thing remaining was to see how the solution behaved under load, as a complex lock-free multi-threaded solution was likely to produce noticeable bugs under stress.

The load test didn't go well, as data stopped coming in at merely a few hundred requests per second. Further observations showed that the 
```C#
await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false)
```

call, which was supposed to return after 1 second, was only returning after 10, even 15 seconds!

I was a sad, sad panda, kind of like this one:
![image](https://user-images.githubusercontent.com/8517601/61569058-66bd0700-aa3a-11e9-96e4-cd4139f1300a.png)

I spent many sleepless hours investigating thread pools and thread priorities, finally arriving at the simple yet efficient solution with a dedicated thread.

I had to spend the next couple of weeks trying to convince SDK people that I really did need a dedicated thread, that 
```C#
Thread.Sleep(TimeSpan.FromSeconds(1))
```
did _not_ consume resources while sleeping, and that the mere fact of allocating a _Thread_ object didn't do that either to any significant degree.

Keep the dedicated thread.

Save the whales.

Reply:
Also, regarding the original .NET Core implementation - it didn't use Threads, but it used a flag which guaranteed that the task would be scheduled on a separate thread. That's why it worked. I believe @netgh0st talked to an expert about it.

Reply:
Yeah that occurred to me as well at the time, but in practice it was the way I described. Maybe HttpClient uses I/O completion ports/threads which have separate handling, maybe there's a separate thread pool... I tested for regular .NET Framework.

Reply:
I guess i will keep the current implementation as is, given that lot of research was done to arrive at original decision, and its not in current scope to re-validate earlier design,
I will abandon this and resubmit a PR which will address a single issue - handle all exception from Thread to ensure we dont crash due to unhandled exceptions.

Thanks all for sharing insights :)